core/database_api.php

```diff
@@ -528,11 +528,21 @@ function db_fetch_array( IteratorAggregate &$p_result ) {
 		return false;
 	}
 
+	# cache these checks to avoid repeated api calls
+	static $t_db_is_pgsql = null;
+	static $t_db_is_oracle = null;
```

Comment:
Static vars must have `s_` prefix.

I would tend to agree with @vboctor on using a single variable here, I don't think it's more complicated to have `$s_db_type == DB_TYPE_xxx` vs a boolean check. Furthermore, since this is about performance optimization, the single-var+constants approach allows use of a switch statement instead of a series of *elseif*, so the code requires fewer tests: 1 to set the static var, and 1 for the check, instead of 2n (so 4 currently).